src/UNL/Peoplefinder/Developers/AbstractResource.php

```diff
@@ -0,0 +1,53 @@
+<?php
+abstract class UNL_PeopleFinder_Developers_AbstractResource
+{
+    const FORMAT_JSON = 'json';
+    const FORMAT_XML = 'xml';
+    const FORMAT_PARTIAL = 'partial';
+    
+    /**
+     * @return string - a brief description of the resource
+     */
+    abstract public function getTitle();
+    
+    /**
+     * @return string - a brief description of the resource
+     */
+    abstract public function getDescription();
+
+    /**
+     * @return mixed - an associative array of property=>description
+     */
+    abstract public function getAvailableFormats();
+
+    /**
+     * @return array - an associative array of property=>description
+     */
+    abstract public function getJsonProperties();
+
+    /**
+     * @return array - an associative array of property=>description
+     */
+    abstract public function getXmlProperties();
```

Comment:
Personally, I'm against this type of setup. I _know_ that historically the output formats have returned different properties, _BUT_ they shouldn't. I shouldn't be able to get more/less data by using a different data interchange format (save partial as it isn't a data interchange format).

What are you thoughts?